tests/unit/customizations/s3/test_fileinfo.py

```diff
@@ -30,18 +29,38 @@ class TestSaveFile(unittest.TestCase):
     def setUp(self):
         self.tempdir = tempfile.mkdtemp()
         self.filename = os.path.join(self.tempdir, 'dir1', 'dir2', 'foo.txt')
-        etag = md5()
-        etag.update(b'foobar')
-        etag = etag.hexdigest()
+        etag = '3858f62230ac3c915f300c664312c63f'
         self.response_data = {
             'Body': six.BytesIO(b'foobar'),
             'ETag': '"%s"' % etag,
         }
         self.last_update = datetime.now()
 
+        # Setup MD5 patches
+        self.md5_object = mock.Mock()
+        self.md5_object.hexdigest = mock.Mock(return_value=etag)
+        md5_builder = mock.Mock(return_value=self.md5_object)
```

Comment:
Unless it's more convenient to store these as instance attributes you can also just say:

```
md5_builder = mock.Mock()
md5_builder.return_value.hexdigest.return_value = etag
```

That makes it more clear that you're only storing mocks that you plan on asserting things on later.